### ISSUE TITLE
Move the enum template operators from tables to core

### ISSUE THREAD

**Issue by gearama** [Azure/azure-sdk-for-cpp#5475]
Move he enum template operators out of tables into core , and make updates to the rest of the code to remove the operator implementations from the various places and use the ones in core

**Comment by ahsonkhan** [comment]
The conclusion from the APIView was to move such "flags" enum bitwise operator helpers to Azure::Core since that would impact any service SDK that uses enum flags.

https://apiview.dev/Assemblies/Review/de3cd5b4510341e6ae1221af4525bd82#c5ac96660ac649ee9bd6fccbfc071ff9

We'd want to do this prior to release, to avoid breaking changes.

https://github.com/Azure/azure-sdk-for-cpp/blob/d835e1a3b1970e53ceda94ef1c6d0596cd103d02/sdk/tables/azure-data-tables/inc/azure/data/tables/enum_operators.hpp#L20-L84

cc @LarryOsterman, @RickWinter

**Comment by ahsonkhan** [comment]
These templated enum operators live in Tables atm. We will not be able to add them to core in the future without risking a breaking change.

From my testing, even if we have two headers with templated bitwise operators for enum, **and in two separate namespaces**, they can't coexist together if the end user includes both headers/packages.

`template <class E, class = std::enable_if_t<std::is_enum<E>{}>>`

If we don't move them to core, any future addition in another package/namespace could cause issues.

```text
Error (active)  E0350  more than one operator "&" matches these operands:
function template "E Azure::Data::Tables::operator&(E lhs, E rhs)" (declared at line 41 of "C:\GitHub\Fork\azure-sdk-for-cpp\sdk\tables\azure-data-tables\inc\azure\data\tables\enum_operators.hpp")
function template "E Azure::Core::operator&(E lhs, E rhs)" (declared at line 47 of "C:\GitHub\Fork\azure-sdk-for-cpp\sdk\tables\azure-data-tables\inc\azure\data\tables\models.hpp")
```

If we try to move it to core later, let's say we have similar operators in the `Azure::Core` namespace, anyone who references both namespaces in their app will get an error/conflict. And operator overloads with templates are quite challenging and verbose to disambiguate/fully qualify.

```C++
using namespace Azure::Data::Tables;
using namespace Azure::Core;

constexpr auto val1 = Azure::Data::Tables::operator|(TestEnum::One, TestEnum::Two);
// vs
constexpr auto val = TestEnum::One | TestEnum::Two;
```

An alternative to moving these templated enum operators to core, would be to add type-specific enum operators, following the precedence from the storage SDK as described here:
https://github.com/Azure/azure-sdk-for-cpp/issues/6153

**Comment by gearama** [comment]
NO, not going to do, attempted it a while back and caused mare confusion than needed, i opened it i am closing it

**Comment by ahsonkhan** [comment]
The conclusion from the API Review was that these generalized/templated enum operators would move to core, and it was only a point-in-time reason why they started off in Tables. I explained above what usability problems an app developer could face if we start introducing multiple templated enum operators in different SDK clients, rather than cerntralizing it.

If we think moving these to core is untenable at this time, **a viable alternative solution is to make these operators specific to the particular enums like we do in Storage SDK**. Is that easier to do? That also resolves the usability concern.

https://github.com/Azure/azure-sdk-for-cpp/blob/c168d736dddeaf0e589ba9fc1c415184c000715c/sdk/storage/azure-storage-blobs/inc/azure/storage/blobs/rest_client.hpp#L441-L477

We should consider making such a change.

**Comment by ahsonkhan** [comment]
Here's the godbolt link to showcase the `ambiguous overload` compiler error in isolation:
https://godbolt.org/z/zKqWeGdTz

**Comment by gearama** [comment]
make the enums non templated and leave them in tables
